src/client/datascience/jupyter/jupyterExecution.ts

```diff
@@ -264,8 +265,8 @@ export class JupyterExecutionBase implements IJupyterExecution {
             this.logger.logWarning(e);
 
             // Double check our jupyter server is still running.
-            if (connection && connection.localProcExitCode) {
-                throw new Error(localize.DataScience.jupyterServerCrashed().format(connection.localProcExitCode.toString()));
+            if (sessionManager && sessionManager.getConnInfo().localProcExitCode) {
+                throw new Error(localize.DataScience.jupyterServerCrashed().format(sessionManager!.getConnInfo().localProcExitCode!.toString()));

```

Comment:
I get that we had to put these due to typescript. But makes me wonder why. Does the actual call to Error happen after this function has exited? Why would the type system not get that you have already checked sessionManager?

Reply:
Don't have the foggiest. Seems broken to me. Maybe it can't parse inside the format function or know that it's not async.

---
In reply to: [321378520](https://github.com/IanMatthewHuff/vscode-python/pull/7#discussion_r321378520) [](ancestors = 321378520)